zetaclient/config/types.go

```diff
@@ -118,12 +121,19 @@ func (c Config) GetAllEVMConfigs() map[int64]EVMConfig {
 	return copied
 }
 
-// GetBTCConfig returns the BTC config
-func (c Config) GetBTCConfig() (BTCConfig, bool) {
+// GetBTCConfig returns the BTC config for the given chain ID
+func (c Config) GetBTCConfig(chainID int64) (BTCConfig, bool) {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 
-	return c.BitcoinConfig, c.BitcoinConfig != (BTCConfig{})
+	// we prefer 'BTCChainConfigs' over 'BitcoinConfig' but still fallback to be backward compatible
+	// this will allow new 'zetaclientd' binary to work with old config file
+	btcCfg, found := c.BTCChainConfigs[chainID]
+	if !found || btcCfg.Empty() {
+		btcCfg = c.BitcoinConfig
```

Comment:
maybe derive the chain ID from `RPCParams` so it's not accidentally mismatched?

Reply:
We actually can't figure out chain ID from the `RPCParams`. Also, the only source of truth would need to come from the RPC endpoint if there IS a way to figure out.

Reply:
Ok yeah I think this logic is fine